src/cmd/cli/command/compose.go

```diff
@@ -298,13 +304,17 @@ func makeComposeDownCmd() *cobra.Command {
 				return err
 			}
 
+			provider, err = canIUseProvider(cmd.Context(), provider, projectName)
+			if err != nil {
+				return err
+			}
+
 			since := time.Now()
 			etag, err := cli.ComposeDown(cmd.Context(), projectName, client, provider, args...)
 			if err != nil {
 				if connect.CodeOf(err) == connect.CodeNotFound {
 					// Show a warning (not an error) if the service was not found
 					term.Warn(prettyError(err))
-					return nil
```

Comment:
Why was this changed? If you `down` a project that's already down, it was considered success, cf. idempotency.